src/python_testing/TC_RVCRUNM_1_2.py

```diff
@@ -0,0 +1,115 @@
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+# This test requires several additional command line arguments
+# run with
+# --int-arg PIXIT_ENDPOINT:<endpoint>
+
+
+class TC_RVCRUNM_1_2(MatterBaseTest):
+
+    async def read_mod_attribute_expect_success(self, endpoint, attribute):
+        cluster = Clusters.Objects.RvcRunMode
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attribute)
+
+    @async_test_body
+    async def test_TC_RVCRUNM_1_2(self):
+
+        asserts.assert_true('PIXIT_ENDPOINT' in self.matter_test_config.global_test_params,
+                            "PIXIT_ENDPOINT must be included on the command line in "
+                            "the --int-arg flag as PIXIT_ENDPOINT:<endpoint>")
+
+        self.endpoint = self.matter_test_config.global_test_params['PIXIT_ENDPOINT']
+
+        attributes = Clusters.RvcRunMode.Attributes
+
+        self.print_step(1, "Commissioning, already done")
+
+        if self.check_pics("RVCRUNM.S.A0000"):
+            self.print_step(2, "Read SupportedModes attribute")
+            supported_modes = await self.read_mod_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.SupportedModes)
+
+            logging.info("SupportedModes: %s" % (supported_modes))
+
+            asserts.assert_greater_equal(len(supported_modes), 2, "SupportedModes must have at least two entries!")
+
+            modes = []
+            for m in supported_modes:
+                if m.mode in modes:
+                    asserts.fail("SupportedModes must have unique mode values!")
+                else:
+                    modes.append(m.mode)
+
+            labels = []
+            for m in supported_modes:
+                if m.label in labels:
+                    asserts.fail("SupportedModes must have unique mode label values!")
+                else:
+                    labels.append(m.label)
+
+            # common mode tags
+            commonTags = {0x0: 'Auto',
+                          0x1: 'Quick',
+                          0x2: 'Quiet',
+                          0x3: 'LowNoise',
+                          0x4: 'LowEnergy',
+                          0x5: 'Vacation',
+                          0x6: 'Min',
+                          0x7: 'Max',
+                          0x8: 'Night',
+                          0x9: 'Day'}
+
+            runTags = [tag.value for tag in Clusters.RvcRunMode.Enums.ModeTag
+                       if tag is not Clusters.RvcRunMode.Enums.ModeTag.kUnknownEnumValue]
+
+            for m in supported_modes:
+                for t in m.modeTags:
+                    is_mfg = (0x8000 <= t.value and t.value <= 0xBFFF)
+                    asserts.assert_true(t.value in commonTags.keys() or t.value in runTags or is_mfg,
+                                        "Found a SupportedModes entry with invalid mode tag value!")
+
```

Comment:
There's something in the test plan about a cleaning mode tag - 
Verify that at least one ModeOptionsStruct entry includes the Cleaning(0x4001) mode tag in the ModeTags field

Verify that none of the ModeOptionsStruct entries include both the Idle(0x4000) mode tag and the Cleaning(0x4001) mode tag in the ModeTags field

I think those checks might be missing.